### ISSUE TITLE
 Duplicate class com.facebook.jni.

### ISSUE THREAD

**Issue by ahmadmssm** [facebook/react-native#34328]
### Description

Hi,

I have added a library to my native module then I tried to run my application but got this error
`
error Failed to install the app. Make sure you have the Android development environment set up: https://reactnative.dev/docs/environment-setup.
Error: Command failed: ./gradlew app:installDebug -PreactNativeDevServerPort=8081

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:checkDebugDuplicateClasses'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.CheckDuplicatesRunnable
   > Duplicate class com.facebook.jni.CppException found in modules jetified-fbjni-0.3.0-runtime (com.facebook.fbjni:fbjni:0.3.0) and jetified-fbjni-java-only-0.2 (com.facebook.fbjni:fbjni-java-only:0.2.2)
     Duplicate class com.facebook.jni.CppSystemErrorException found in modules jetified-fbjni-0.3.0-runtime (com.facebook.fbjni:fbjni:0.3.0) and jetified-fbjni-java-only-0.2 (com.facebook.fbjni:fbjni-java-only:0.2.2)
     Duplicate class com.facebook.jni.DestructorThread found in modules jetified-fbjni-0.3.0-runtime (com.facebook.fbjni:fbjni:0.3.0) and jetified-fbjni-java-only-0.2 (com.facebook.fbjni:fbjni-java-only:0.2.2)
     Duplicate class com.facebook.jni.DestructorThread$1 found in modules jetified-fbjni-0.3.0-runtime (com.facebook.fbjni:fbjni:0.3.0) and jetified-fbjni-java-only-0.2 (com.facebook.fbjni:fbjni-java-only:0.2.2)
     Duplicate class com.facebook.jni.DestructorThread$Destructor found in modules jetified-fbjni-0.3.0-runtime (com.facebook.fbjni:fbjni:0.3.0) and jetified-fbjni-java-only-0.2 (com.facebook.fbjni:fbjni-java-only:0.2.2)
     Duplicate class com.facebook.jni.DestructorThread$DestructorList found in modules jetified-fbjni-0.3.0-runtime (com.facebook.fbjni:fbjni:0.3.0) and jetified-fbjni-java-only-0.2 (com.facebook.fbjni:fbjni-java-only:0.2.2)
     Duplicate class com.facebook.jni.DestructorThread$DestructorStack found in modules jetified-fbjni-0.3.0-runtime (com.facebook.fbjni:fbjni:0.3.0) and jetified-fbjni-java-only-0.2 (com.facebook.fbjni:fbjni-java-only:0.2.2)
     Duplicate class com.facebook.jni.DestructorThread$Terminus found in modules jetified-fbjni-0.3.0-runtime (com.facebook.fbjni:fbjni:0.3.0) and jetified-fbjni-java-only-0.2 (com.facebook.fbjni:fbjni-java-only:0.2.2)
     Duplicate class com.facebook.jni.HybridClassBase found in modules jetified-fbjni-0.3.0-runtime (com.facebook.fbjni:fbjni:0.3.0) and jetified-fbjni-java-only-0.2 (com.facebook.fbjni:fbjni-java-only:0.2.2)
     Duplicate class com.facebook.jni.HybridData found in modules jetified-fbjni-0.3.0-runtime (com.facebook.fbjni:fbjni:0.3.0) and jetified-fbjni-java-only-0.2 (com.facebook.fbjni:fbjni-java-only:0.2.2)
     Duplicate class com.facebook.jni.HybridData$Destructor found in modules jetified-fbjni-0.3.0-runtime (com.facebook.fbjni:fbjni:0.3.0) and jetified-fbjni-java-only-0.2 (com.facebook.fbjni:fbjni-java-only:0.2.2)
     Duplicate class com.facebook.jni.IteratorHelper found in modules jetified-fbjni-0.3.0-runtime (com.facebook.fbjni:fbjni:0.3.0) and jetified-fbjni-java-only-0.2 (com.facebook.fbjni:fbjni-java-only:0.2.2)
     Duplicate class com.facebook.jni.MapIteratorHelper found in modules jetified-fbjni-0.3.0-runtime (com.facebook.fbjni:fbjni:0.3.0) and jetified-fbjni-java-only-0.2 (com.facebook.fbjni:fbjni-java-only:0.2.2)
     Duplicate class com.facebook.jni.NativeRunnable found in modules jetified-fbjni-0.3.0-runtime (com.facebook.fbjni:fbjni:0.3.0) and jetified-fbjni-java-only-0.2 (com.facebook.fbjni:fbjni-java-only:0.2.2)
     Duplicate class com.facebook.jni.ThreadScopeSupport found in modules jetified-fbjni-0.3.0-runtime (com.facebook.fbjni:fbjni:0.3.0) and jetified-fbjni-java-only-0.2 (com.facebook.fbjni:fbjni-java-only:0.2.2)
     Duplicate class com.facebook.jni.UnknownCppException found in modules jetified-fbjni-0.3.0-runtime (com.facebook.fbjni:fbjni:0.3.0) and jetified-fbjni-java-only-0.2 (com.facebook.fbjni:fbjni-java-only:0.2.2)
     Duplicate class com.facebook.jni.annotations.DoNotStrip found in modules jetified-fbjni-0.3.0-runtime (com.facebook.fbjni:fbjni:0.3.0) and jetified-fbjni-java-only-0.2 (com.facebook.fbjni:fbjni-java-only:0.2.2)
     `

### Version

react-native-cli: 2.0.1 / react-native: 0.68.2

### Output of `npx react-native info`

Duplicate class com.facebook.jni.CppException found in modules jetified-fbjni-0.3.0-runtime 

### Steps to reproduce

Just run ` npx yarn example android stacktrace`

### Snack, code example, screenshot, or link to a repository

Private Repo.

**Comment by cortinico** [comment]
> I have added a library to my native module then I tried to run my application but got this error

What is happening is that more than one library is importing FBJNI, and you're ending up with duplicate classes. 

> Private Repo.

Without a repro it's hard to say where is the problem. At least having a build.gradle file of your app to look into would be beneficial

**Comment by ahmadmssm** [comment]
@cortinico
I have excluded FBJNI from the library like this:
```
implementation ("myLibrary") {
    exclude group:'com.facebook.fbjni', module: 'fbjni-java-only'
}
```  
What is weird is that my library is not using Facebook SDK at all.

**Comment by cortinico** [comment]
> What is weird is that my library is not using Facebook SDK at all.

FBJNI is **not** the Facebook SDK: https://github.com/facebookincubator/fbjni

Your library is depending on either Flipper or React Native or some other libraries which depend on `fbjni`.

**Comment by ahmadmssm** [comment]
@cortinico 

Yes, my library is using Flipper which depends on `fbjni` as you mentioned, so what I did in the gradle file is to exclude `fbjni` like this:

```
implementation ("MY_LIBRARY") {
    exclude group:'com.facebook.fbjni'
}
```

But it didn't work until I added this line 
` exclude group:'com.facebook.flipper'`

After adding the mentioned line, It finally worked.

**Comment by cortinico** [comment]
> After adding the mentioned line, It finally worked.

Closing then 👍

**Comment by alija98** [comment]
@ahmadmssm Hello, I have the same error as you. How did you know which library uses Flipper and where to exclude it?

**Comment by alija98** [comment]
> @ahmadmssm Hello, I have the same error as you. How did you know which library uses Flipper and where to exclude it?

I have made Stackoverflow question https://stackoverflow.com/questions/74344054/duplicate-class-com-facebook-jni with my android/build.gradle and android/app/build.gradle

**Comment by omarkhaled11** [comment]
I have the same issue, using the above solutions like disabling Flipper didn't work. 
also the builds were successful before the weekend, no external libraries were added since then, the error just appeared out of no where on CI and on our local machines.

**Comment by Mehdi4924** [comment]
> `exclude group:'com.facebook.flipper'`

Yes, I'm facing the same from 2 days, does any body got a solution for this?

**Comment by dgreasi** [comment]
Same here

**Comment by dgreasi** [comment]
> This is happening since November 4th. New React Native update makes android builds not work for older versions. Here you can find the solution, just instead of your RN version in package.json change it to the patched one, it worked for me -> #35210

It worked for my case

**Comment by cortinico** [comment]
If you're seeing this issue, starting from Nov 4th, 
The fix for this issue is the one listed here:
https://github.com/facebook/react-native/issues/35210
as already suggested.
